src/app/utilities/moodle/helper.ts

```diff
@@ -0,0 +1,175 @@
+  }
+}
+
+export async function createMoodleUser({
+  email,
+  name,
+}: {
+  email: string;
+  name: string;
+}): Promise<MoodleUser> {
+  try {
+    const splitName = name.split(/\s+/);
+    const firstname = splitName.slice(0, -1).join(' ');
+    const lastname = splitName.slice(-1).join('');
+    const username = email;
+
+    const password = 'TestPass123!' + getUniqueID();
+
+    const formData = new FormData();
+    formData.append('wstoken', MOODLE_SECRET);
+    formData.append('wsfunction', 'core_user_create_users');
+    formData.append('moodlewsrestformat', 'json');
+    formData.append('users[0][email]', email);
+    formData.append('users[0][firstname]', firstname);
+    formData.append('users[0][lastname]', lastname);
+    formData.append('users[0][password]', password);
+    formData.append('users[0][username]', username);
+    formData.append('users[0][auth]', 'oidc');
+
+    const res = await fetch(MOODLE_API_URL, {
+      method: 'POST',
+      body: formData,
+    });
+
+    const json: MoodleUser[] | MoodleException = await res.json();
+    if ('exception' in json) {
+      throw new Error(json.message);
+    }
+
+    const user: MoodleUser = json[0];
+
+    return user;
+  } catch (_) {
+    throw new Error('Failed to create the moodle user.');
+  }
+}
+
+export async function enrolMoodleUserInCourse({
+  userid,
+  courseid,
+}: {
+  userid: number;
+  courseid: number;
+}): Promise<void> {
+  try {
+    const formData = new FormData();
+    formData.append('wstoken', MOODLE_SECRET);
+    formData.append('wsfunction', 'enrol_manual_enrol_users');
+    formData.append('moodlewsrestformat', 'json');
+    formData.append('enrolments[0][roleid]', MOODLE_STUDENT_ROLE_ID.toString());
+    formData.append('enrolments[0][userid]', userid.toString());
+    formData.append('enrolments[0][courseid]', courseid.toString());
+
+    const res = await fetch(MOODLE_API_URL, {
+      method: 'POST',
+      body: formData,
+    });
+    const json: MoodleException | null = await res.json();
+
+    if (json && 'exception' in json) {
+      throw new Error(json.message);
+    }
+  } catch (_) {
+    throw new Error('Failed to enrol the moodle user into the course.');
+  }
+}
+
+export async function getMoodleCoursesByUser(
+  userid: number
+): Promise<MoodleCourse[]> {
+  try {
+    const formData = new FormData();
+    formData.append('wstoken', MOODLE_SECRET);
+    formData.append('wsfunction', 'core_enrol_get_users_courses');
+    formData.append('moodlewsrestformat', 'json');
+    formData.append('userid', userid.toString());
+
+    const res = await fetch(MOODLE_API_URL, {
+      method: 'POST',
+      body: formData,
+    });
+    const json: MoodleCourse[] | MoodleException = await res.json();
+
+    if ('exception' in json) {
+      throw new Error(json.message);
+    }
+
+    return json;
+  } catch (_) {
+    throw new Error(`Failed to fetch the user's enrolled courses.`);
+  }
+}
```

Comment:
It seems like the code to call moodle is repetitive, we are changing the data and method name. We can isolate this to reduce duplication.